core/arch/arm/kernel/trace_ext.c

```diff
@@ -28,20 +28,39 @@
 #include <trace.h>
 #include <console.h>
 #include <kernel/thread.h>
+#include <kernel/spinlock.h>
+#include <mm/core_mmu.h>
 
 const char trace_ext_prefix[] = "TEE-CORE";
 int trace_level = TRACE_LEVEL;
+static unsigned int puts_lock = SPINLOCK_UNLOCK;
 
 void trace_ext_puts(const char *str)
 {
+	uint32_t itr_status = thread_mask_exceptions(THREAD_EXCP_FOREIGN_INTR);
```

Comment:
@jforissier : can you crosscheck, from SW archi view, that this "foreign exception only masking" will not lead to a deadlock in case of a native interruption handler that output some trace (same about async aborts ending in some OP-TEE handler that would generate a trace) ?
Even if such setup is likely to be a debug setup, deadlocks are not enjoyable to anyone.